### PR TITLE
fix(core): Support CIFS permission restrictions in source control

### PR DESCRIPTION
**Pull request by ivov**
CIFS filesystems [disallow changing file permissions after mount](https://sysadmins.tech/linux-and-cifs-files-permissions/), hence set the permissions for the SSH private key on creation instead of using `chmod`.

Context: https://n8nio.slack.com/archives/C04B1GZ4T0U/p1726665721479049

### CODE REVIEW

**Review by netroy — DISMISSED** [review]

**Comment by ivov** [comment]
Will wait for customer to confirm before merging.

**Comment by netroy** [comment]
if this is not ready to merge, we should mark this as back in draft and dismiss the review.

**Comment by ivov** [comment]
Closing, let's reopen if/when the customer reports back

**Pull request closed without merging**
